src/libsystemd/sd-journal/mmap-cache.c

```diff
@@ -278,41 +277,49 @@ static int add_mmap(
         assert(size > 0);
         assert(ret);
 
-        woffset = offset & ~((uint64_t) page_size() - 1ULL);
-        wsize = size + (offset - woffset);
-        wsize = PAGE_ALIGN(wsize);
+        /* PAGE_ALIGN() and friends used below take size_t. If we pass too large value these may provide
+         * an unexpected result. */
+        if (offset > (uint64_t) SIZE_MAX)
```

Comment:
Shouldn't we then fix the macros so they can work with `size_t`? It makes sense to me that we'd support journals larger than 4Gib even on 32-bit systems.

Reply:
OK. Pushed another version. PTAL.